pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -365,14 +380,77 @@ func (s *streamIngestionResumer) handleResumeError(
 // Resume is part of the jobs.Resumer interface.  Ensure that any errors
 // produced here are returned as s.handleResumeError.
 func (s *streamIngestionResumer) Resume(resumeCtx context.Context, execCtx interface{}) error {
+	// Protect the destination tenant's keyspan from garbage collection.
+	err := s.protectDestinationTenant(resumeCtx, execCtx)
+	if err != nil {
+		return s.handleResumeError(resumeCtx, execCtx, err)
+	}
+
 	// Start ingesting KVs from the replication stream.
-	err := ingestWithRetries(resumeCtx, execCtx.(sql.JobExecContext), s.job)
+	err = ingestWithRetries(resumeCtx, execCtx.(sql.JobExecContext), s.job)
 	if err != nil {
 		return s.handleResumeError(resumeCtx, execCtx, err)
 	}
 	return nil
 }
 
+func releaseDestinationTenantProtectedTimestamp(
+	ctx context.Context, execCtx interface{}, txn *kv.Txn, ptsID uuid.UUID,
+) error {
+	jobExecCtx := execCtx.(sql.JobExecContext)
+	ptp := jobExecCtx.ExecCfg().ProtectedTimestampProvider
+	if err := ptp.Release(ctx, txn, ptsID); err != nil {
+		if errors.Is(err, protectedts.ErrNotExists) {
+			// No reason to return an error which might cause problems if it doesn't
+			// seem to exist.
```

Comment:
If we return an error here then that will bubble up to `OnFailOrCancel` which is a method the job registry does not allow to fail. In other words, we will retry `OnFailOrCancel` endlessly (with backoff) until it succeeds which it will never since it won't be able to find the record. A reason why we might not find the record is if `OnFailOrCancel` gets retried causing us to invoke this method more than once in separate txns. The expectation is that all operations in `Resume` and `OnFailOrCancel` should be idempotent and so we safeguard against multiple invocations.

Reply:
oh this definitely makes sense in that case. thanks for explaining!